### PR TITLE
Fix "Create New Section" dialog no-scrolling bug

### PR DESCRIPTION
**Pull request by TurnerRiley**
Fixes a scrolling bug in the New Section dialog window where LoginTypeCards were getting cut off since it wasn't scrolling. The original problem was spotted when [this PR](https://github.com/code-dot-org/code-dot-org/pull/40384) was merged. 

### Case when the user does not have a 3rd party provider connected to their account (e.g. Google Classroom) so they only have access to Picture Password, Secret Words, and Personal Login types (only 1 row of 3 LoginTypeCards):
Fix scroll bug 1
### Case when the user has at least one 3rd party provider connected to their account (e.g. Google Classroom) so they have at least 4 types of login cards (causing 2 rows of cards):
Fix scroll bug 2

### Case when the user does not have a 3rd party provider connected to their account (e.g. Google Classroom) so they only have access to Picture Password, Secret Words, and Personal Login types (only 1 row of 3 LoginTypeCards):
Fix scroll bug 3
### Case when the user has at least one 3rd party provider connected to their account (e.g. Google Classroom) so they have at least 4 types of login cards (causing 2 rows of cards):
Fix scroll bug 4

## Testing story
Manual testing with different numbers of LoginTypeCards (by manually toggling whether I have third party options enabled or not such as Google Classrooms) to ensure correct scrolling behavior when there is not enough space to display all of the content at once. This testing was also replicated in multiple browsers while ensuring that the full process of going through the motions of creating a new section still work (from clicking "New section" to the actual creation of a new section). Lastly, I created an ad-hoc environment to triple check its functionality.

### CODE REVIEW

**Review by daynew — APPROVED** [review]
LGTM, thank you for finding an elegant solution.

File: apps/src/templates/teacherDashboard/LoginTypePicker.jsx [hunk 1]
```diff
@@ -79,18 +99,18 @@ class LoginTypePicker extends Component {
...
-        {!hasThirdParty && (
-          <div>
-            {i18n.thirdPartyProviderUpsell() + ' '}
-            <a href="https://support.code.org/hc/en-us/articles/115001319312-Setting-up-sections-with-Google-Classroom-or-Clever">
-              {i18n.learnHow()}
-            </a>
-            {' ' + i18n.connectAccountThirdPartyProviders()}
-          </div>
-        )}
-        <DialogFooter>
```
Comment (annaxuphoto) [inline apps/src/templates/teacherDashboard/LoginTypePicker.jsx:92]: We don't need the DialogFooter anymore?

**Review by annaxuphoto — APPROVED** [review]
This looks good to me, just one question in comments. Thanks for the thorough screenshots and testing story :) In the future, maybe a screen recording would be easier!

File: apps/src/templates/teacherDashboard/LoginTypePicker.jsx (hunk 1 above)
Comment (TurnerRiley) [reply apps/src/templates/teacherDashboard/LoginTypePicker.jsx:92]: @annaxuphoto Thank you! I'll make sure to attach recordings when that fits better! The DialogFooter has specific absolute positioning styling that was causing issues with the scrolling bug, and I didn't want to modify the DialogFooter itself to mess up other implementations of DialogFooter.

**Pull request merged**